rust/core-lib/src/movement.rs

```diff
@@ -248,8 +248,11 @@ pub fn region_movement(
                 } else if cursor.pos() == text.len() {
                     offset = text.len();
                 }
+                (offset, None)
+            } else {
```

Comment:
In this case, we can just do `offset = text.len()`. Not sure if we can clean up the logic any, so that we don't have two branches with the same result?

Reply:
Looks like the easiest way to keep code clean is just to use `(text.len(), None)`